streams/src/main/java/org/apache/kafka/streams/state/internals/CachingWindowStore.java

```diff
@@ -160,6 +160,9 @@ public synchronized void put(final Bytes key, final byte[] value, final long tim
         validateStoreOpen();
         final Bytes bytesKey = WindowKeySchema.toStoreKeyBinary(key, timestamp, 0);
         final Bytes cacheKey = cacheFunction.cacheKey(bytesKey);
+        if (cache == null) {
```

Comment:
In `CachingSessionStore#findSessions`, we could also check if `cache` is null.

Reply:
Looking at CachingSessionStore#findSessions, an Iterator is returned.
```
        final ThreadCache.MemoryLRUCacheBytesIterator cacheIterator = cache.range(cacheName, cacheKeyFrom, cacheKeyTo);
```
When cache is null, should a special MemoryLRUCacheBytesIterator be returned ?